pkg/util/limiter/query_limiter.go

```diff
@@ -28,7 +28,7 @@ var (
 
 type QueryLimiter struct {
 	uniqueSeriesMx sync.Mutex
-	uniqueSeries   map[uint64]struct{}
+	uniqueSeries   map[uint64]*labels.Labels
```

Comment:
Why does this need to be a pointer?

Reply:
Initially I am thinking this is the way to prevent duplicated allocation.

Reply:
Updated the map value into non-pointer. Added some test to check whether the labels are using the same instance for duplicated series.